### ISSUE TITLE
Array.new() bug?

### ISSUE THREAD

**Issue by konung** [ruby2js/ruby2js#147]
Hi. 

Just trying out ruby2js with various random ruby idioms, ran into an output that confused me. I'm not sure, if it's a bug per se, because it depends on what the goal of Ruby2Js when it says it “does conversion”.

Ruby:
```ruby
Array.new(5, 1).map{|x| x + 2 }
# [3, 3, 3, 3, 3]
```

Produces this js

```javascript
new Array(5, 1).map(x => x + 2)
// Array [ 7, 3 ]
```

So I'm wondering should I expect ruby2js to produce JavaScript code that generates the same output? Or should I expect ruby2js to produce valid JS and I need to examine, if it does, what I would expect equivalent code in Ruby to produce?

In other words, what does “conversion” mean in the context of the project : 
Is ruby2js a compiler or a code generator?

P.S.:
Sorry if I'm asking something obvious, just trying to understand the goal of the project 😄
